### PR TITLE
 Remove duplicated +1 in datetime range calculations & misc fixes

### PR DESCRIPTION
**Pull request by ihnorton**

### CODE REVIEW

**Review by nguyenv — APPROVED** [review]

File: tiledb/libtiledb.pyx [hunk 1]
```diff
@@ -2620,10 +2620,10 @@ cdef class Dim(object):
                     # this means to use the full extent of the type
                     if dtype.kind == 'M':
                         date_unit = np.datetime_data(dtype)[0]
-                        dtype_min = np.datetime64(dtype_min + 1, date_unit)
+                        dtype_min = np.datetime64(dtype_min, date_unit)
```
Comment (gsakkis) [inline tiledb/libtiledb.pyx:2623]: Isn't this expression what `dtype_range` does to compute `dtype_min`? I think it's not needed anymore.

File: tiledb/libtiledb.pyx [hunk 2]
```diff
@@ -2620,10 +2620,10 @@ cdef class Dim(object):
...
                     elif dtype == np.int64 and not (dtype.kind == 'M'):
                         # except that the domain range currently must fit in UInt64
-                        dtype_min = dtype_min + 1
+                        dtype_min = dtype_min
```
Comment (gsakkis) [inline tiledb/libtiledb.pyx:2626]: This is redundant now.

So with the comment above, I think the whole block can be simplified to:
```
if domain == (None, None):
    domain = (dtype_min, dtype_max)
````

File: tiledb/libtiledb.pyx (hunk 1 above)
Comment (ihnorton) [reply tiledb/libtiledb.pyx]: ✔️

File: tiledb/libtiledb.pyx (hunk 2 above)
Comment (ihnorton) [reply tiledb/libtiledb.pyx]: Thanks

**Review by gsakkis — APPROVED** [review]

**Pull request merged**